privacyidea/api/lib/prepolicy.py

```diff
@@ -2688,3 +2688,21 @@ def rss_age(request, action):
             log.warning(f"Invalid RSS_AGE: {age_list}. Using the default.")
     request.all_data[ACTION.RSS_AGE] = age
     return True
+
+
+def disabled_token_types(request, action):
+    """
+    This decorator retrieves the disabled token types from the policies and adds them to the request data,
+    to disable them for the authentication in check_token_list.
+
+    :param request: The request object
+    :param action: The action parameter is not used in this decorator
+    :return: True
+    """
+    disabled = Match.user(g, scope=SCOPE.AUTH, action=ACTION.DISABLED_TOKEN_TYPES,
+                             user_object=request.User if hasattr(request, 'User') else None).action_values(unique=False)
+
+    if disabled:
+        request.all_data[ACTION.DISABLED_TOKEN_TYPES] = list(disabled)
```

Comment:
Maybe we should also *remove* a given entry from `all_data` if the policy isn't set.

Reply:
just set an emtpy list